gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/simple/AbstractClient.java

```diff
@@ -37,11 +39,16 @@
  */
 public abstract class AbstractClient implements SimpleClient {
     protected final CallbackResponseHandler callbackResponseHandler = new CallbackResponseHandler();
-    protected final EventLoopGroup group;
+    public final EventLoopGroup group;
+    // Checks and uses Epoll if it is available. ref: http://netty.io/wiki/native-transports.html
+    // Subclasses also depend on if Epoll is available or not, so making this protected
+    protected boolean isEpollAvailable;
```

Comment:
I'm not quite convinced this is necessary as it's just a wrapper around Epoll.isAvailable().

Or, in other words, what is kind of concerning to me in this PR is that we are calling Epoll.isAvailable() in many places, but then again it looks like the logic of what to do after checking it is difference so having this consolidated in one place doesn't even seem really feasible.

Reply:
I guess there isn't any difference, so you mean to just use Epoll.isAvailable() directly?

Reply:
Yeah.

Like, it would make sense to wrap it if we had like a function that returns a EventLoopGroup pool and it abstracted this away, but it looks like we still end up needing to call Epoll.isAvailable() directly anyways so it's pointless.